tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/PropagateDeleteMailboxRightListener.scala

```diff
@@ -0,0 +1,88 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.team
+
+import com.google.common.annotations.VisibleForTesting
+import com.linagora.tmail.team.PropagateDeleteMailboxRightListener.GROUP
+import com.linagora.tmail.team.TeamMailboxNameSpace.TEAM_MAILBOX_NAMESPACE
+import jakarta.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.events.EventListener.ReactiveGroupEventListener
+import org.apache.james.events.{Event, Group}
+import org.apache.james.mailbox.MailboxManager
+import org.apache.james.mailbox.acl.ACLDiff
+import org.apache.james.mailbox.events.MailboxEvents.MailboxACLUpdated
+import org.apache.james.mailbox.model.MailboxACL.EntryKey
+import org.apache.james.mailbox.model.{MailboxACL, MailboxPath}
+import org.reactivestreams.Publisher
+import org.slf4j.LoggerFactory
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+case class PropagateDeleteMailboxRightListenerGroup() extends Group {}
+
+object PropagateDeleteMailboxRightListener {
+  val GROUP: PropagateDeleteMailboxRightListenerGroup = PropagateDeleteMailboxRightListenerGroup()
+}
+
+class PropagateDeleteMailboxRightListener @Inject()(teamMailboxRepository: TeamMailboxRepository,
+                                                    mailboxManager: MailboxManager) extends ReactiveGroupEventListener {
+
+  private val logger = LoggerFactory.getLogger(classOf[PropagateDeleteMailboxRightListener])
+
+  override def getDefaultGroup: Group = GROUP
+
+  override def isHandling(event: Event): Boolean = event.isInstanceOf[MailboxACLUpdated]
```

Comment:
Note: 

Use `MailboxACLUpdated` instead of `MailboxAdded` to prevent race condition issues with `StoreMailboxManager#inheritRightsReactive`.
The `PropagateLookupRightListener` will run after `inheritRights` is completed.